pkg/base-dev/lib/sss.hoon

```diff
@@ -0,0 +1,280 @@
+/-  *sss
+::
+|%
+++  mk-subs  |*  [=(lake) paths=mold]  -:+6:(da lake paths)
+++  mk-pubs  |*  [=(lake) paths=mold]  -:+6:(du lake paths)
+++  mk-mar
+  |*  =(lake)
+  |_  =(response:poke lake *)
+  ++  grow
+    |%
+    ++  noun  response
+    --
+  ++  grab
+    |%
+    ++  noun  (response:poke lake *)
+    --
+  ++  grad  %noun
+  --
+++  lake-mark
+  |=  =mark
+  ^+  mark
+  ?>  =('sss-' (end [3 4] mark))
+  (cut 3 [4 (met 3 mark)] mark)
+::
+++  fled                                     ::  Like +sped but head is a path.
+  |=  vax=vase
+  ^-  vase
+  :_  q.vax
+  %-  ~(play ut p.vax)
+  =-  [%wtgr [%wtts - [%& 2]~] [%$ 1]]
+  =/  pax  ~|  %need-path  ;;(path -.q.vax)
+  |-  ^-  spec
+  ?~  pax  [%base %null]
+  [%bccl ~[[%leaf %ta -.pax] $(pax +.pax)]]
+::
+++  zoom  |=  =noun  ~|  %path-none  $/sss/;;(path noun)
+++  da
+  |*  [=(lake) paths=mold]
+  =>
+    |%
+    +$  flow
+      $:  =aeon
+          rok=[=aeon fail=_| =rock:lake]
+          wav=((mop aeon wave:lake) lte)
+      ==
+    --
+  |_  [sub=(map [ship dude paths] flow) =bowl:gall result-type=type on-rock-type=type]
+  +*  wav     ((on aeon wave:lake) lte)
+  +$  from    (on-rock:poke lake paths)
+  +$  into    (response:poke lake paths)
+  +$  result  (request:poke paths)
+  ++  pine
+    |=  [=what =ship =dude path=paths]
+    ^-  card:agent:gall
+    :*  %pass   (zoom request/pine/ship^dude^what^path)
+        %agent  [ship dude]
+        %poke   %sss-to-pub  :-  result-type  ^-  result
+        [path dap.bowl %pine what]
+    ==
+  ++  surf
+    |=  [=ship =dude path=paths]
+    (pine %wave ship dude path)
+  ::
+  ++  read
+    ;;  (map [ship dude path] [fail=? rock:lake])
+    %-  ~(run by sub)
+    |=  =flow
+    [fail rock]:rok.flow
+  ::
+  ++  chit
+    |=  [[aeon=term ship=term dude=term path=paths] =sign:agent:gall]
+    ^+  sub
+    ?>  ?=(%poke-ack -.sign)
+    ?~  p.sign  sub
+    %+  ~(jab by sub)  [(slav %p ship) dude path]
+    |=  =flow
+    ?.  =(aeon.rok.flow (slav %ud aeon))  flow
```

Comment:
silent no-op here seems dangerous to me -- at least print, maybe crash

Reply:
It crashes now.